src/main/java/org/victorrobotics/frc/dtlib/log/DTLogger.java

```diff
@@ -0,0 +1,202 @@
+package org.victorrobotics.frc.dtlib.log;
+
+import org.victorrobotics.frc.dtlib.DTLibInfo;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.util.WPILibVersion;
+
+public class DTLogger {
+  private static final DateTimeFormatter TIME_FORMATTER = //
+      DateTimeFormatter.ofPattern("uuuu-MM-dd_HH-mm-ss")
+                       .withZone(ZoneId.of("Z"));
+
+  private static final String DTLIB_LOGGER_MAGIC = "DTLib Logger";
+  private static final String LOG_DIRECTORY      = "/dev/sda/logs";
+
+  static final Map<Class<?>, DTLogType<?>> LOGGABLE_TYPES = new HashMap<>();
+
+  static {
+    DTLogBuiltinTypes.load();
+  }
+
+  private static File        file;
+  private static DTLogWriter dataWriter;
+
+  private static long lastTimestamp;
+  private static long startTimeMillis;
+  private static int  nextHandle = 0x0100;
+
+  private DTLogger() {}
+
+  public static DTLogWriter getWriter() {
+    return dataWriter;
+  }
+
+  public static int newHandle(int typeID, String path) {
+    dataWriter.writeShort(typeID);
+    dataWriter.writeStringUTF8(path);
+    return nextHandle++;
+  }
+
+  public static boolean logNewTimestamp() {
+    long newTime = RobotController.getFPGATime() / 1000;
+    long diff = newTime - lastTimestamp;
+    if (diff <= 0) {
+      // Same time, no change
+      return false;
+    }
+    lastTimestamp = newTime;
+
+    if (diff <= 0xFFFF) {
+      // 2 byte form, maximum of 65.535 seconds
```

Comment:
I think we should consider always writing a 4-byte timestamp, because the short written before it makes the actual lengths 4 bytes and 6 bytes.